### DISCUSSION TITLE
Organisation Entity into ALM

### DISCUSSION THREAD

**Discussion by StylezDevops** [microsoft/powerplatform-build-tools#450] [Ideas]
Hi Team!!
Not  a bug - but as discussed on our call @kkanakas 

We have the ability to Create Environments with the build Tools... I would like to see a task that would replicate the CHANGED settings from one environment to other environments.

I see it working like so: Developers make any changes required to their development environment - making changes to the System Settings dialogue (organisation entity). As a DevOps engineer i would the ability  "export settings" from this developer environment and to apply these same settings to my newly created environments before deploying solutions to them.

I currently do this in PowerShell and leverage the Web API to query my dev environment for settings - which I then apply (again with a bit of PowerShell) to other environments. 

This ensures consistency across all of our environments when developing and deploying solutions - would be great to see this in an official task and is not something that would be difficult to implement.

Thanks and Kind Regards
Dan Styles

**Comment by StuartBaxter-MVP** [comment]
Having worked alongside Dan on projects, I know firsthand how useful his approach to transferring org settings is.

Having a command in PAC cli would be super useful for PP admins

**Comment by Laskewitz** [comment]
@StylezDevops - This sounds a lot like #407, with the difference that this is a copy functionality. 

`pac org copy-settings --sourceEnvironment <SOURCE ENV ID> --destinationEnvironment <DESTINATION ENV ID>` should do the trick. And then of course a wrapper around that for a DevOps task / GitHub action.

Edit: 
It would be nice to have `--sourceEnvironment` as an optional parameter, if not applied, then use the currently selected environment.
